### ISSUE TITLE
Improved Table.read compatibility with np.savetxt output

### ISSUE THREAD

**Issue by parejkoj** [astropy/astropy#8674]
`np.savetxt` [defaults to](https://docs.scipy.org/doc/numpy/reference/generated/numpy.savetxt.html) prepending a `#` on all header lines:

```
comments : str, optional
String that will be prepended to the header and footer strings, to mark them as comments. Default: ‘# ‘, as expected by e.g. numpy.loadtxt
```

`numpy.genfromtxt` claims that it will by default ignore all such lines:

```
comments : str, optional (default='#')
The character used to indicate the start of a comment. All the characters occurring on a line after a comment are discarded
```

but if you specify `names=True` to tell it to read the column names from the header, it will strip that leading `# `:

```
names : {None, True, str, sequence}, optional (default=None)
If names is True, the field names are read from the first line after the first skip_header lines. This line can optionally be proceeded by a comment delimiter.  ...
```

However, if such a `np.savetxt`ed file is read with `Table.read(format='csv')`, the first column will be named `# name`, instead of `name`.

It would be useful to have an option in astropy's CSV reader to be able to strip leading comment lines from the header, so that `np.savetxt` output can be read directly. Something like `ignore_leading_header_character="#"` (not the best name choice, I know). Numpy's documentation on the interaction of the `comments` and `names` fields is not great, but based on a quick skim of the numpy issues related to these methods, it's probably easier to provide a workaround in astropy.

**Comment by taldcroft** [comment]
Are you looking for the `commented_header` format?
```
In [11]: Table.read(['# a b', '1 2'], format='ascii.commented_header')
Out[11]: 
<Table length=1>
  a     b  
int64 int64
----- -----
    1     2
```
If not, please provide a direct example of the data file you want to read.

**Comment by parejkoj** [comment]
Ah-hah! That does it exactly. Thank you.

Worth adding a mention to the docs for that format that it matches what `np.savetxt` produces?

**Comment by bsipocz** [comment]
ahh, your `format='csv'` totally tricked me of not remembering this. Sorry.

**Comment by bsipocz** [comment]
This has been addressed in #8694
